src/unix/fs.c

```diff
@@ -48,7 +48,8 @@
 #if defined(__DragonFly__)  ||                                            \
     defined(__FreeBSD__)    ||                                            \
     defined(__OpenBSD__)    ||                                            \
-    defined(__NetBSD__)
+    defined(__NetBSD__)     ||                                            \
+    defined(__linux__)
 # define HAVE_PREADV 1
 #else
 # define HAVE_PREADV 0
```

Comment:
I know... is that an acceptable tradeoff? Kernels <= 2.6.30 are from 2009, and the oldest still maintained "LTS" kernel is 2.6.32. What systems would be break?